cpp/src/gandiva/gdv_function_stubs.cc

```diff
@@ -332,8 +332,8 @@ const char* gdv_fn_aes_encrypt(int64_t context, const char* data, int32_t data_l
     *out_len = gandiva::aes_encrypt(data, data_len, key_data,
                                     reinterpret_cast<unsigned char*>(ret));
   } catch (const std::runtime_error& e) {
-    gdv_fn_context_set_error_msg(context, e.what());
-    return nullptr;
+    std::string err_msg = "Following error occurred while encrypting ciphertext - " + std::string(e.what());
+    gdv_fn_context_set_error_msg(context, err_msg.data());
```

Comment:
Since this falls through to returning 'ret', is there a chance that ret would be invalid? In other words, I'm wondering what cases return a valid 'ret' pointer, but the aes_encrypt call fails?

Reply:
In this case of a runtime exception (which occurs when wrong decryption key is given), a valid 'ret' is returned, but the error set on context is correctly shown on UI. Returning null pointer instead of this, results in a segmentation fault.